tests/unit/layers/models/layersManager.spec.ts

```diff
@@ -0,0 +1,717 @@
+      validateGpkgFilesMock.mockReturnValue(true);
+      createLayerJobMock.mockResolvedValue('testJobId');
+      createMergeTilesTasksMock.mockResolvedValue(undefined);
+
+      // const zoomLevelCalculator = new ZoomLevelCalculator(configMock);
+      // layersManager = new LayersManager(
+      //   logger,
+      //   zoomLevelCalculator,
+      //   jobManagerClientMock,
+      //   catalogClientMock,
+      //   mapPublisherClientMock,
+      //   fileValidatorMock,
+      //   splitTilesTaskerMock,
+      //   mergeTilesTasker
+      // );
+
+      await layersManager.createLayer(testData, managerCallbackUrl);
+
+      expect(getHighestLayerVersionMock).toHaveBeenCalledTimes(1);
+      expect(fileValidatorValidateExistsMock).toHaveBeenCalledTimes(1);
+      expect(findJobsMock).toHaveBeenCalledTimes(1);
+      expect(validateGpkgFilesMock).toHaveBeenCalledTimes(1);
+      expect(createMergeTilesTasksMock).toHaveBeenCalledTimes(1);
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment
+      const mockId = createMergeTilesTasksMock.mock.calls[0][0].metadata.id;
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment
+      const mockDisplayPath = createMergeTilesTasksMock.mock.calls[0][0].metadata.displayPath;
+      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+      const relativePath = `${mockId}/${mockDisplayPath}`;
+      expect(createMergeTilesTasksMock).toHaveBeenCalledWith(
+        testData,
+        relativePath,
+        TaskAction.MERGE_TILES,
+        JobAction.NEW,
+        [Grid.TWO_ON_ONE],
+        [0, 0, 1, 1],
+        managerCallbackUrl,
+        true
+      );
+    });
+
+    it('createMergeTilesTasksMock function should not called with "isNew" parameter for "Update" job type', async function () {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      setValue({ 'tiling.zoomGroups': '1,2-3' });
+      setValue('ingestionTilesSplittingTiles.tasksBatchSize', 2);
+      const testData: IngestionParams = {
+        fileNames: ['test.gpkg'],
+        metadata: { ...testImageMetadata },
+        originDirectory: '/here',
+      };
+
+      const getGridSpy = jest.spyOn(SQLiteClient.prototype, 'getGrid');
+      getGridSpy.mockReturnValue(Grid.TWO_ON_ONE);
+      getHighestLayerVersionMock.mockResolvedValue(2.0);
+      fileValidatorValidateExistsMock.mockResolvedValue(true);
+      validateSourceDirectoryMock.mockResolvedValue(true);
+      validateNotWatchDirMock.mockResolvedValue(true);
+      mapExistsMock.mockResolvedValue(true);
+      findJobsMock.mockResolvedValue([]);
+      validateGpkgFilesMock.mockReturnValue(true);
+      createLayerJobMock.mockResolvedValue('testJobId');
+      createMergeTilesTasksMock.mockResolvedValue(undefined);
+
+      // const zoomLevelCalculator = new ZoomLevelCalculator(configMock);
```

Comment:
why the remarks?

Reply:
moved into before each test instad of create new instance every test manualy